### PR TITLE
Add SBL support for kernelflinger

### PR DESCRIPTION
**Pull request by GangSecurity**
1. Add efiwrapper and libpayload library to support SBL
2. Reuse kerneflinger.c for SBL entry
3. Solve compile issues

Tracked-On: OAM-110183

### CODE REVIEW

**Review by dyang23 — APPROVED** [review]
LGTM

File: Android.mk [hunk 1]
```diff
@@ -280,6 +284,7 @@ LOCAL_STATIC_LIBRARIES := \
 	libfastboot-for-installer-$(TARGET_BUILD_VARIANT) \
 	libxbc-$(TARGET_BUILD_VARIANT)
 
+
```
Comment (James-Zzz) [inline Android.mk:287]: remove abundant blank line

File: Android.mk [hunk 2]
```diff
@@ -389,8 +395,13 @@ LOCAL_STATIC_LIBRARIES += libavb_kernelflinger-$(TARGET_BUILD_VARIANT)
 ifneq ($(TARGET_BUILD_VARIANT),user)
     LOCAL_STATIC_LIBRARIES += libadb-$(TARGET_BUILD_VARIANT)
 endif
+#Replace kf4abl.c by kernelflinger.c
```
Comment (James-Zzz) [inline Android.mk:398]: Why do we need to replace kf4abl.c by kernelflinger.c?
Does boot logic change with this replacement?

File: libkernelflinger/android.c [hunk 3]
```diff
@@ -1281,15 +1368,33 @@ static EFI_STATUS setup_command_line(
...
 #ifdef AUTO_DISKBUS
+#ifdef USE_SBL
+		int32_t bdf;
+		uint32_t diskbus1, storage_bus_num;
+		diskbus1 = get_diskbus();
+
+		debug(L"nvme controller diskbus = %x\n", diskbus1);
+		storage_bus_num = diskbus1>>16;
+
+		bdf = bridge_diskbus (storage_bus_num);
```
Comment (James-Zzz) [inline libkernelflinger/android.c:1379]: bridge_diskbus(storage_bus_num), remove abundant space

File: libkernelflinger/android.c [hunk 4]
```diff
@@ -1281,15 +1368,33 @@ static EFI_STATUS setup_command_line(
         if (boot_device) {
                 CHAR16 *diskbus = NULL;
 #ifdef AUTO_DISKBUS
+#ifdef USE_SBL
+		int32_t bdf;
+		uint32_t diskbus1, storage_bus_num;
+		diskbus1 = get_diskbus();
```
Comment (James-Zzz) [inline libkernelflinger/android.c:1374]: How about rename "diskbus1" to "disk_bus_num", since "diskbus1" is easily to get confused with "diskbus"

**Review by kwang13 — APPROVED** [review]

**Review by James-Zzz — APPROVED** [review]
LGTM

**Review by James-Zzz — APPROVED** [review]
LGTM

**Pull request closed without merging**